### ISSUE TITLE
Tree: Default value for checkbox

### ISSUE THREAD

**Issue by zanyar3** [primefaces/primevue#3837]
### Describe the feature you would like to see added

To set a default value for the Tree component with checkbox selection mode, you can initialize the `selectedKey `data property with an array of the keys that correspond to the nodes you want to be selected by default.
For example, if you want to select the nodes with keys "node1" and "node2" by default, you can set `selectedKey` to `["node1","node2"]` in the component's data object:

### Example
```
<template>
  <Tree
    v-model:selectionKeys="selectedKey"
    :value="nodes"
    :filter="true"
    selectionMode="checkbox"
    :metaKeySelection="true"
    class="w-full"
    style="min-width: 22rem"
  />
</template>

<script setup>
const  selectedKey = ref(["node1", "node2"]);
const nodes = [
        { key: "node1", label: "Node 1" },
        { key: "node2", label: "Node 2" },
        { key: "node3", label: "Node 3" }
      ];
</script>
```

In this example, the `selectedKey `array contains the keys of the **"Node 1" and "Node 2"** nodes, so they will be selected by default when the component is mounted. You can replace these keys with the keys of the nodes you want to be selected by default

### Describe alternatives you have considered

You can useing `selectedKey` for set default value.
**Example for set default-value**

```
const selectedKey = ref({
  'node1': {
    checked: true,
    partialChecked: false,
  },
});
```

by using code can fixed set default value but my problem with `partialChecked`, not easy control is `true `or `false `and need more code for provider it.

**Comment by tugcekucukoglu** [comment]
Tree component represents hierarchical data as you said and its `selectionKeys` property keeps the map of keys to represent the expansion state. So this is a map, not an array.

**Comment by sohepalslamat** [comment]
> Tree component represents hierarchical data as you said and its `selectionKeys` property keeps the map of keys to represent the expansion state. So this is a map, not an array.

Hello @tugcekucukoglu,

I wanted to let you know that I have created a suggestion to fix the problem we've been experiencing with. The suggested fix is in this PR #3853. I have tested this solution thoroughly and am confident that it will address the issue.

I have submitted a pull request with the proposed changes and would greatly appreciate it if you could review and provide feedback on the suggested fix. If you have any questions or concerns about the proposed solution, please do not hesitate to reach out to me.

Thank you for your attention to this matter.

**Comment by zanyar3** [comment]
Hi,

Alternative for fixing the problem you can use the Custom function to generate keys.

```
const nodes = getNodesFromAnyWehere();
const defaultValues = [1, 3]
const keys = generateKeysByIds(nodes , defaultValues);
```

###  Set ref variable value for tree component
`selectedKey.value = keys ;`


### custom function
```
function generateKeysByIds(data, ids) {
  const output = {};

  function traverse(items, prefix) {
    items.forEach((item) => {
      const key = item.key;

      const checked = ids.includes(item.id);
      const partialChecked = item.children && item.children.some((child) => ids.includes(child.id));

      if (checked || partialChecked) {
        output[key] = {
          checked: checked,
          partialChecked: partialChecked,
        };
      }

      if (item.children && item.children.length > 0) {
        traverse(item.children, key);
      }
    });
  }

  traverse(data, "");

  return output;
}
```
